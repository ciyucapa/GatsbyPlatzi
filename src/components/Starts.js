import React, {useState} from 'react';

import {SelectStars} from '../styles/components'

const Starts = () => {
    const [stars, setStars] = useState(5)
    return (
        <SelectStars selected={stars}>
            <span role="button" tabIndex="0" onClick={() => setStars(1)} onKeyDown={() => setStars(1)}>★</span>
            <span role="button" tabIndex="0" onClick={() => setStars(2)} onKeyDown={() => setStars(2)}>★</span>
            <span role="button" tabIndex="0" onClick={() => setStars(3)} onKeyDown={() => setStars(3)}>★</span>
            <span role="button" tabIndex="0" onClick={() => setStars(4)} onKeyDown={() => setStars(4)}>★</span>
            <span role="button" tabIndex="0" onClick={() => setStars(5)} onKeyDown={() => setStars(5)}>★</span>
        </SelectStars>
    )
}

export default Starts