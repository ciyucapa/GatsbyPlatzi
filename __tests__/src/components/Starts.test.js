import React from "react"
import renderer from "react-test-renderer"
import Starts from '../../../src/components/Starts'

describe("Starts", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(<Starts 
                    stars={5} 
                    onStars={() => 5} 
                />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    });
});