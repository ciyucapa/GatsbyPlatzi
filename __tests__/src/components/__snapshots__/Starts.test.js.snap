// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Starts renders correctly 1`] = `
<div
  className="sc-VigVT bLIVrL"
  selected={5}
>
  <span
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  >
    ★
  </span>
  <span
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  >
    ★
  </span>
  <span
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  >
    ★
  </span>
  <span
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  >
    ★
  </span>
  <span
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex="0"
  >
    ★
  </span>
</div>
`;
